python-package/lightgbm/basic.py

```diff
@@ -2126,6 +2126,8 @@ def _lazy_init(
                 categorical_feature=categorical_feature,
                 pandas_categorical=self.pandas_categorical,
             )
+        elif _is_pyarrow_table(data):
```

Comment:
```suggestion
        elif isinstance(str, feature_name) and feature_name == "auto" and _is_pyarrow_table(data):
```

As currently written, your patch would overwrite whatever was passed in via the `feature_name` keyword argument. We only want that behavior when `feature_name="auto"`, just as with `pandas`:

https://github.com/microsoft/LightGBM/blob/e0c34e7b2f68793bbf46854b85fd750be134446d/python-package/lightgbm/basic.py#L840-L842

Please ensure this use of `data.column_names` only happens when `feature_name="auto"`, and add a unit test case confirming that.

Reply:
Ah phew, I think I misinterpreted how `feature_name` is supposed to work in my original Arrow-related PRs 🤔 I think this PR fixes a broader issue then :smile: passing `feature_name` has always been ignored for PyArrow tables, right?

Reply:
Yeah I think you're right, I never noticed this either!

But there are legitimate reasons for wanting to use the `feature_name` argument instead of automatic feature names from the columns. For example, if you want have very-long feature names and want to reduce the size of the model file. Of if your feature names in the source data contain sensitive information that you don't want to be encoded in the model.

Those are slightly niche and "just change the names in the input data" would be a fair response to them... but since we've supported that with `pandas` for a while I think we should support it for all dataframe-like inputs.

Reply:
> Those are slightly niche and "just change the names in the input data" would be a fair response to them... but since we've supported that with pandas for a while I think we should support it for all dataframe-like inputs.

I agree! Given the current behavior, we just adapt the PyArrow table support to do the same as pandas.